spyder/config/utils.py

```diff
@@ -99,7 +99,16 @@ def _get_pygments_extensions():
             lexer_exts = [le for le in lexer_exts if not le.endswith('_*')]
             extensions = extensions + list(lexer_exts) + list(other_exts)
 
-    return sorted(list(set(extensions)))
+    extensions = list(set(extensions))
+
+    # A non-ascii file extension causes issues for macOS
+    # See spyder-ide/spyder#22248
+    try:
+        extensions.remove('.' + chr(128293))
```

Comment:
Just in case, seems like this extension is related with `mojo` [which allows using as extension the fire emoji (🔥)](https://docs.modular.com/mojo/faq#why-does-mojo-have-the--file-extension). The emoji even appears in [the pygments docs for lexers](https://pygments.org/docs/lexers/#pygments.lexers.mojo.MojoLexer).

Not sure if is worthy to leave a comment related to that in the code though

Reply:
Actually, this was already mentioned over the issue so I would say this has enough traceability :+1:(https://github.com/spyder-ide/spyder/issues/22248#issuecomment-2237849663)